desktop-src/direct3dhlsl/AddUint64.md

```diff
@@ -0,0 +1,61 @@
+---
+description: Unsigned add of 32-bit operand with the carry.
+nms.assetid:
+title: AddUint64
+ms.topic: reference
+ms.date: 07/10/2024
+topic_type:
+- APIRef
+- kbSyntax
+api_name:
+- AddUint64
+api_type:
+- NA
+---
+
+
+# AddUint64
+
+Unsigned add of 32-bit operand with the carry.
+
+## Syntax
+
+
+```syntax
+uint<c> AddUint64(uint<c> a, uint<c> b);
+```
+
+## Parameters
+
+| Item | Description |
+|------|-------------|
+| *a* | [in] A vector of type uint64_t (64-bit unsigned integer) as the first operand.  || *b* | [in] A second vector of type uint64_t (64-bit unsigned integer) as the second operand.  |
+## Return value
+
+ Returns a vector resulting from the addition of the two input vectors. Each component in the resulting vector is the sum of the corresponding components of the input vectors. All operations are performed on uint64_t types.
+## Type Description
+
+| Name  | [**Template Type**](../direct3dhlsl/dx-graphics-hlsl-data-types.md)| [**Component Type**](../direct3dhlsl/dx-graphics-hlsl-data-types.md) | Size |
+|-------|--------------------------------------------------------------------|----------------------------------------------------------------------|------|
+| *ret*   | [**vector**](../direct3dhlsl/dx-graphics-hlsl-vector.md) | [**uint**](../WinProg/windows-data-types) | any |
+| *a*   | [**vector**](../direct3dhlsl/dx-graphics-hlsl-vector.md) | [**uint**](../WinProg/windows-data-types) | any |
+| *b*   | [**vector**](../direct3dhlsl/dx-graphics-hlsl-vector.md) | [**uint**](../WinProg/windows-data-types) | any |
```

Comment:
Size should be 2 or 4 here (See "AddUint64 can only be applied to uint2 and uint4 operands" error in HLOperationLower.cpp)

Reply:
we don't encode that in `DirectXShaderCompiler.utils.hct.hctdb_instrhelp` so my view is just that it is a `uint<c>` which mean this is something that would have to be a manual edit since `hctdb_instrhelp` isn't giving enough detail.